celery/backends/redis.py

```diff
@@ -138,6 +141,14 @@ def cancel_for(self, task_id):
             self.subscribed_to.discard(key)
             self._pubsub.unsubscribe(key)
 
+    @property
```

Comment:
If I understand the proposal of @thedrow :

```python
@property
def pubsub(self):
    if self._thread.pubsub is None:
        # Create pubsub client instance here
        self._thread.pubsub = pubsub_instance
    return self._thread.pubsub 
```
Thus the setter is not required.

Reply:
I don't think we should make `_pubsub` public but otherwise, @georgepsarakis is correct.